### ISSUE TITLE
Cipher `TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384` is not available when TCP input with SSL is used "before" beats input with SSL

### ISSUE THREAD

**Issue by jakommo** [logstash-plugins/logstash-input-beats#316]
- Version: 6.2.3
- Operating System: Ubuntu 16.04
- Config File (if you have sensitive info, please remove it): See blow

From https://discuss.elastic.co/t/logstash-tls-with-x-pack-error-registering-plugin-cipher-is-not-available-6-x/117650/8 and another User reporting. (I can also reproduce)

If both, tcp and beats input are configured with SSL, and tcp input comes first in the config file, it's causing a `LogStash::ConfigurationError: Cipher `TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384` is not available` error.

- Steps to Reproduce:
```
 jakommo@es-jre  ~/testing  cat tcp_first.conf 
input {
	tcp {
		port => 5011
		ssl_enable => true
		ssl_cert => "/home/jakommo/testing/ssl-cert-snakeoil.pem"
		ssl_key => "/home/jakommo/testing/ssl-cert-snakeoil.key"
	}
	beats {
		port => 5010
		ssl => true
		ssl_certificate => "/home/jakommo/testing/ssl-cert-snakeoil.pem"
		ssl_key => "/home/jakommo/testing/ssl-cert-snakeoil.key"
	}
}
 jakommo@es-jre  ~/testing  ~/TOOLS/logstash-6.2.3/bin/logstash -f tcp_first.conf  
Sending Logstash's logs to /home/jakommo/TOOLS/logstash-6.2.3/logs which is now configured via log4j2.properties
[2018-04-16T17:27:11,516][INFO ][logstash.modules.scaffold] Initializing module {:module_name=>"fb_apache", :directory=>"/home/jakommo/TOOLS/logstash-6.2.3/modules/fb_apache/configuration"}
[2018-04-16T17:27:11,548][INFO ][logstash.modules.scaffold] Initializing module {:module_name=>"netflow", :directory=>"/home/jakommo/TOOLS/logstash-6.2.3/modules/netflow/configuration"}
[2018-04-16T17:27:12,625][WARN ][logstash.config.source.multilocal] Ignoring the 'pipelines.yml' file because modules or command line options are specified
[2018-04-16T17:27:14,143][INFO ][logstash.runner          ] Starting Logstash {"logstash.version"=>"6.2.3"}
[2018-04-16T17:27:14,990][INFO ][logstash.agent           ] Successfully started Logstash API endpoint {:port=>9600}
[2018-04-16T17:27:19,263][INFO ][logstash.pipeline        ] Starting pipeline {:pipeline_id=>"main", "pipeline.workers"=>4, "pipeline.batch.size"=>125, "pipeline.batch.delay"=>50}
[2018-04-16T17:27:19,491][INFO ][logstash.inputs.tcp      ] Starting tcp input listener {:address=>"0.0.0.0:5011", :ssl_enable=>"true"}
[2018-04-16T17:27:20,471][INFO ][logstash.inputs.beats    ] Beats inputs: Starting input listener {:address=>"0.0.0.0:5010"}
[2018-04-16T17:27:20,751][ERROR][logstash.pipeline        ] Error registering plugin {:pipeline_id=>"main", :plugin=>"<LogStash::Inputs::Beats port=>5010, ssl=>true, ssl_certificate=>\"/home/jakommo/testing/ssl-cert-snakeoil.pem\", ssl_key=>\"/home/jakommo/testing/ssl-cert-snakeoil.key\", id=>\"4dd14fd2f978bbc8c548d4d753f8726cae13e4a704d8aefe013762cef769fff4\", enable_metric=>true, codec=><LogStash::Codecs::Plain id=>\"plain_df06f0be-4c16-439d-8018-bedae0bcd07a\", enable_metric=>true, charset=>\"UTF-8\">, host=>\"0.0.0.0\", ssl_verify_mode=>\"none\", include_codec_tag=>true, ssl_handshake_timeout=>10000, tls_min_version=>1, tls_max_version=>1.2, cipher_suites=>[\"TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384\", \"TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384\", \"TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256\", \"TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256\", \"TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA384\", \"TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA384\", \"TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256\", \"TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256\"], client_inactivity_timeout=>60, executor_threads=>4>", :error=>"Cipher `TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384` is not available", :thread=>"#<Thread:0x3ebc1874 run>"}
[2018-04-16T17:27:20,968][ERROR][logstash.pipeline        ] Pipeline aborted due to error {:pipeline_id=>"main", :exception=>#<LogStash::ConfigurationError: Cipher `TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384` is not available>, :backtrace=>["/home/jakommo/TOOLS/logstash-6.2.3/vendor/bundle/jruby/2.3.0/gems/logstash-input-beats-5.0.10-java/lib/logstash/inputs/beats.rb:170:in `create_server'", "/home/jakommo/TOOLS/logstash-6.2.3/vendor/bundle/jruby/2.3.0/gems/logstash-input-beats-5.0.10-java/lib/logstash/inputs/beats.rb:158:in `register'", "/home/jakommo/TOOLS/logstash-6.2.3/logstash-core/lib/logstash/pipeline.rb:341:in `register_plugin'", "/home/jakommo/TOOLS/logstash-6.2.3/logstash-core/lib/logstash/pipeline.rb:352:in `block in register_plugins'", "org/jruby/RubyArray.java:1734:in `each'", "/home/jakommo/TOOLS/logstash-6.2.3/logstash-core/lib/logstash/pipeline.rb:352:in `register_plugins'", "/home/jakommo/TOOLS/logstash-6.2.3/logstash-core/lib/logstash/pipeline.rb:502:in `start_inputs'", "/home/jakommo/TOOLS/logstash-6.2.3/logstash-core/lib/logstash/pipeline.rb:393:in `start_workers'", "/home/jakommo/TOOLS/logstash-6.2.3/logstash-core/lib/logstash/pipeline.rb:289:in `run'", "/home/jakommo/TOOLS/logstash-6.2.3/logstash-core/lib/logstash/pipeline.rb:249:in `block in start'"], :thread=>"#<Thread:0x3ebc1874 run>"}
[2018-04-16T17:27:21,041][ERROR][logstash.agent           ] Failed to execute action {:id=>:main, :action_type=>LogStash::ConvergeResult::FailedAction, :message=>"Could not execute action: LogStash::PipelineAction::Create/pipeline_id:main, action_result: false", :backtrace=>nil}
 ✘ jakommo@es-jre  ~/testing 

            
 ✘ jakommo@es-jre  ~/testing  cat beats_first.conf 
input {
	beats {
		port => 5010
		ssl => true
		ssl_certificate => "/home/jakommo/testing/ssl-cert-snakeoil.pem"
		ssl_key => "/home/jakommo/testing/ssl-cert-snakeoil.key"
	}
	tcp {
		port => 5011
		ssl_enable => true
		ssl_cert => "/home/jakommo/testing/ssl-cert-snakeoil.pem"
		ssl_key => "/home/jakommo/testing/ssl-cert-snakeoil.key"
	}
}
 jakommo@es-jre  ~/testing  ~/TOOLS/logstash-6.2.3/bin/logstash -f beats_first.conf 
Sending Logstash's logs to /home/jakommo/TOOLS/logstash-6.2.3/logs which is now configured via log4j2.properties
[2018-04-16T17:28:21,516][INFO ][logstash.modules.scaffold] Initializing module {:module_name=>"fb_apache", :directory=>"/home/jakommo/TOOLS/logstash-6.2.3/modules/fb_apache/configuration"}
[2018-04-16T17:28:21,570][INFO ][logstash.modules.scaffold] Initializing module {:module_name=>"netflow", :directory=>"/home/jakommo/TOOLS/logstash-6.2.3/modules/netflow/configuration"}
[2018-04-16T17:28:22,578][WARN ][logstash.config.source.multilocal] Ignoring the 'pipelines.yml' file because modules or command line options are specified
[2018-04-16T17:28:23,916][INFO ][logstash.runner          ] Starting Logstash {"logstash.version"=>"6.2.3"}
[2018-04-16T17:28:24,882][INFO ][logstash.agent           ] Successfully started Logstash API endpoint {:port=>9600}
[2018-04-16T17:28:28,809][INFO ][logstash.pipeline        ] Starting pipeline {:pipeline_id=>"main", "pipeline.workers"=>4, "pipeline.batch.size"=>125, "pipeline.batch.delay"=>50}
[2018-04-16T17:28:29,482][INFO ][logstash.inputs.beats    ] Beats inputs: Starting input listener {:address=>"0.0.0.0:5010"}
[2018-04-16T17:28:29,613][INFO ][logstash.inputs.tcp      ] Starting tcp input listener {:address=>"0.0.0.0:5011", :ssl_enable=>"true"}
[2018-04-16T17:28:30,280][INFO ][logstash.pipeline        ] Pipeline started succesfully {:pipeline_id=>"main", :thread=>"#<Thread:0x36ab5c5a run>"}
[2018-04-16T17:28:30,482][INFO ][org.logstash.beats.Server] Starting server on port: 5010
[2018-04-16T17:28:30,556][INFO ][logstash.agent           ] Pipelines running {:count=>1, :pipelines=>["main"]}
```

**Comment by jakommo** [comment]
@robbavey just a quick check, looks like the latest version of the plugin is `5.0.14` and not `5.0.4`. 
Is this a typo in your comment? 
I did test with `5.0.4` yesterday and it worked as well, but looking for clarification.

**Comment by robbavey** [comment]
@jakommo - Sorry I misstated the plugin that the fix was made in. The library discrepancy was fixed in version `5.0.4` of the **tcp** input. Any version released after that is ok. The latest version of tcp is `5.0.7` and that should also work correctly.

**Comment by jakommo** [comment]
Got it, thanks for clarifying @robbavey .
